bentoml/_internal/frameworks/mlflow.py

```diff
@@ -184,24 +185,35 @@ def import_model(
         from mlflow.pyfunc import FLAVOR_NAME as PYFUNC_FLAVOR_NAME
         from mlflow.models.model import MLMODEL_FILE_NAME
 
+        # Explicitly provide a destination dir to mlflow so that we don't
+        # accidentially download into the root of the bento model temp dir
+        # (using a model:/ url can cause this)
+        download_dir = tempfile.mkdtemp(dir=bento_model.path)
+
         try:
             # Prefer public API download_artifacts introduced in MLflow 1.25
             from mlflow.artifacts import download_artifacts
 
             local_path = download_artifacts(
-                artifact_uri=model_uri, dst_path=bento_model.path
+                artifact_uri=model_uri, dst_path=download_dir
             )
         except ImportError:
             # For MLflow < 1.25
             from mlflow.tracking.artifact_utils import _download_artifact_from_uri
 
             local_path = _download_artifact_from_uri(
-                artifact_uri=model_uri, output_path=bento_model.path
+                artifact_uri=model_uri, output_path=download_dir
             )
 
         mlflow_model_path = bento_model.path_of(MLFLOW_MODEL_FOLDER)
         # Rename model folder from original artifact name to fixed "mlflow_model"
         shutil.move(local_path, mlflow_model_path)
+        # If the temp dir we created still exists now, we never needed it because
+        # the provided mlflow url must have provided enough path information.  Just
+        # delete it.  If it's not here, it means we needed it but it's been renamed
+        # by now so we don't need to remove it.
+        if os.path.isdir(download_dir):
+            shutil.rmtree(download_dir)
```

Comment:
I'd suggest to put this logic in a `finally` block. We can merge this and add the finally block as a follow-up.